tests/demos/erc_20_token.rs

```diff
@@ -0,0 +1,150 @@
+// This contract is a translation of
+// https://github.com/Pi-Squared-Inc/pi2-examples/blob/b63d0a78922874a486be8a0395a627425fb5a052/solidity/src/tokens/SomeToken.sol
+//
+// The main change is that the contract does not issue specific error objects
+// (e.g. ERC20InsufficientBalance), it just calls `require!` with various
+// (string) explanations.
+//
+// Also, the `totalSupply` endpoint is declared implicitely as a view for
+// `s_total_supply`.
+
+#![no_std]
+
+#[allow(unused_imports)]
+use multiversx_sc::imports::*;
+
+#[multiversx_sc::contract]
+pub trait Erc20Token {
+    #[view(totalSupply)]
+    #[storage_mapper("total_supply")]
+    fn s_total_supply(&self) -> SingleValueMapper<BigUint>;
+
+    #[view(getName)]
+    #[storage_mapper("name")]
+    fn s_name(&self) -> SingleValueMapper<ManagedBuffer>;
+
+    #[view(getSymbol)]
+    #[storage_mapper("symbol")]
+    fn s_symbol(&self) -> SingleValueMapper<ManagedBuffer>;
+
+    #[view(getBalances)]
+    #[storage_mapper("balances")]
+    fn s_balances(&self, address: &ManagedAddress) -> SingleValueMapper<BigUint>;
+
+    #[view(getAllowances)]
+    #[storage_mapper("allowances")]
+    fn s_allowances(&self, account: &ManagedAddress, spender: &ManagedAddress) -> SingleValueMapper<BigUint>;
+
+    #[event("Transfer")]
+    fn transfer_event(&self, #[indexed] from: &ManagedAddress, #[indexed] to: &ManagedAddress, value: &BigUint);
+    #[event("Approval")]
+    fn approval_event(&self, #[indexed] owner: &ManagedAddress, #[indexed] spender: &ManagedAddress, value: &BigUint);
+
+
+    #[init]
+    fn init(&self, name: &ManagedBuffer, symbol: &ManagedBuffer, init_supply: &BigUint) {
+        self.s_name().set_if_empty(name);
+        self.s_symbol().set_if_empty(symbol);
+        self._mint(&self.blockchain().get_caller(), init_supply);
+    }
+
+    #[upgrade]
+    fn upgrade(&self) {}
+
+    #[view(decimals)]
+    fn decimals(&self) -> u8 {
+        18_u8
+    }
+
+    // Already declared above
+    // #[view(totalSupply)]
```

Comment:
The reason why we're keeping this section of commented code is just to highlight this difference between how we achieve what is originally offered in the erc20 Solidity contract, is that the case?

Reply:
Well... It's because the current contract works on the real MX blockchain, but we didn't implement yet views for storage, so it does not work for us. If we don't implement this (and it seems that we will not implement it), a quick fix is to uncomment this code.